.github/workflows/run-cypress-on-pr.yaml

```
@@ -0,0 +1,73 @@
+name: Cypress tests on pr
+on:
+  pull_request:
+    branches: [ master ]
+    types: [opened, synchronize, reopened]
+    paths:
+      - 'frontend/**'
+      - '!frontend/stats/**'
+      - 'backend/**'
+      - '.github/workflows/run-cypress-on-pr.yaml'
+      - 'docker-compose.yml'
+      - 'Dockerfile'
+      - 'gitea/**'
+  workflow_dispatch:
+
+jobs:
+  cypress-tests:
+    name: Build environment and run e2e test
+    timeout-minutes: 25
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+
+      - name: Generate .env file
+        run: |
+          echo CYPRESS_TEST_APP=autodeploy-v3 >> .env
+          echo DEVELOP_APP_DEVELOPMENT=0 >> .env
+          echo DEVELOP_RESOURCE_ADMIN=0 >> .env
+          echo DEVELOP_BACKEND=0 >> .env
+          echo DEVELOP_DASHBOARD=0 >> .env
+          echo DEVELOP_PREVIEW=0 >> .env
+          echo GITEA_ADMIN_PASS=g9wDIG@6gf >> .env
+          echo GITEA_ADMIN_USER=localg1iteaadmin >> .env
+          echo GITEA_CYPRESS_USER=cypress_testuser >> .env
+          echo GITEA_CYPRESS_PASS=g9wDIG@6gf >> .env
+          echo GITEA_ORG_USER=ttd >> .env
+          echo POSTGRES_PASSWORD=kyeDIG@eip >> .env
+          echo COMMIT= >> .env
+          echo IGNORE_DOCKER_DNS_LOOKUP=true >> .env
```

What are your thoughts on adding empty strings as credential, since these will be replaced by the setup.js file? Then it's more clear that the credential properties are generated by the setup.js script.

```suggestion
          echo GITEA_ADMIN_PASS= >> .env
          echo GITEA_ADMIN_USER= >> .env
          echo GITEA_CYPRESS_USER= >> .env
          echo GITEA_CYPRESS_PASS= >> .env
          echo GITEA_ORG_USER=ttd >> .env
          echo POSTGRES_PASSWORD= >> .env
          echo COMMIT= >> .env
          echo IGNORE_DOCKER_DNS_LOOKUP=true >> .env
```

---

Values are not replaced by setup.js. They are actually used by docker-compose files. For example POSTGRES_PASSWORD is used by postress container.